test/e2e/e2e_test.go

```diff
@@ -157,6 +161,83 @@ var _ = ginkgo.Describe("Kueue visibility server", func() {
 				}, util.Timeout, util.Interval).Should(gomega.Equal(0))
 			})
 		})
+
+		ginkgo.It("Should allow fetching information about position of pending workloads", func() {
+			var (
+				info *visibility.PendingWorkloadsSummary
+			)
+
+			ginkgo.By("Schedule a job that when admitted workloads blocks the queue", func() {
+				sampleJob1 = testingjob.MakeJob("test-job", ns.Name).
+					Queue(localQueueA.Name).
+					Image("gcr.io/k8s-staging-perf-tests/sleep:v0.0.3", []string{"5s"}).
+					Request(corev1.ResourceCPU, "1").
+					PriorityClass(highPriorityClass.Name).
+					Obj()
+				gomega.Expect(k8sClient.Create(ctx, sampleJob1)).Should(gomega.Succeed())
+			})
+
+			ginkgo.By("Schedule three different jobs with two different priorities and two different LocalQueues", func() {
+				jobCases := []struct {
+					JobName          string
+					JobPrioClassName string
+					LocalQueueName   string
+				}{
+					{
+						JobName:          "lq-a-high-prio",
+						JobPrioClassName: highPriorityClass.Name,
+						LocalQueueName:   localQueueA.Name,
+					},
+					{
+						JobName:          "lq-b-high-prio",
+						JobPrioClassName: highPriorityClass.Name,
+						LocalQueueName:   localQueueB.Name,
+					},
+					{
+						JobName:          "lq-b-low-prio",
+						JobPrioClassName: lowPriorityClass.Name,
+						LocalQueueName:   localQueueB.Name,
+					},
+				}
+				for _, jobCase := range jobCases {
+					job := testingjob.MakeJob(jobCase.JobName, ns.Name).
+						Queue(jobCase.LocalQueueName).
+						Image("gcr.io/k8s-staging-perf-tests/sleep:v0.0.3", []string{"1s"}).
+						Request(corev1.ResourceCPU, "1").
+						PriorityClass(jobCase.JobPrioClassName).
+						Obj()
+					gomega.Expect(k8sClient.Create(ctx, job)).Should(gomega.Succeed())
+				}
+			})
+
+			ginkgo.By("Verify there are three pending workloads", func() {
+				gomega.Eventually(func() int {
+					var err error
+					info, err = visibilityClient.ClusterQueues().GetPendingWorkloadsSummary(ctx, clusterQueue.Name, metav1.GetOptions{})
+					gomega.Expect(err).NotTo(gomega.HaveOccurred())
+					return len(info.Items)
+				}, util.Timeout, util.Interval).Should(gomega.Equal(3))
+			})
+
+			ginkgo.By("Verify their position in ClusterQueue", func() {
+				gomega.Expect(info.Items[0].PositionInClusterQueue).To(gomega.Equal(0))
```

Comment:
This is hard to read, can we instead use the `ComparableTo` function (or such) to compare the entire list of pending workloads to the expected list?